### PR TITLE
Various OLM related improvements

### PR DESCRIPTION
**Pull request by eriknelson**
* Moves deploy_olm switch out of the role so it can be enabled at the
entrypoint playbook to control whether or not OLM should be deployed.

* Parameterizes olm_namespace so that it can be provided. By default, it
will be dynamically looked up since we've seen this change in the
upstream OLM manifests.

* Only attempts to deploy OLM in 3.11+ installs, since we've seen this
breaking with anything prior in CI.

### CODE REVIEW

**Comment by eriknelson** [comment]
FYI, the admin console is busted due to the OLM namespace rename. It had a hardcoded namespace: https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/index.tsx#L167

From slack: "This hard-coded namespace will be removed once OLM is cluster-wide"

File: ansible/roles/olm_setup/defaults/main.yml [hunk 1]
```diff
@@ -11,6 +10,14 @@ olm_repo: "https://github.com/operator-framework/operator-lifecycle-manager.git"
 olm_repo_dest: /tmp/olm
 olm_manifest_type: okd
 
+# Hack here will force deployment to old operator-lifecycle-manager namespace,
+# which is what the admin console is hardcoded to.
+# # TODO: Remove this once the admin console is fixed!!
+olm_namespace_hack: false
```
Comment (djwhatle) [inline ansible/roles/olm_setup/defaults/main.yml]: Should we default this to true until a fix comes in?

File: ansible/roles/olm_setup/tasks/main.yml [hunk 2]
```diff
@@ -38,6 +41,23 @@
...
+    block:
+    - shell: "grep 'openshift-operator-lifecycle-manager' -lr {{ _olm_manifests}}"
+      register: files_to_patch
+    - shell: "sed -i 's|openshift-operator-lifecycle-manager|operator-lifecycle-manager|' {{ item }}"
+      with_items: "{{ files_to_patch.stdout_lines }}"
+
+  - when: olm_namespace == ""
+    block:
+    - name: Get OLM namespace
+      shell: "perl -ne '/name: (.*)$/ && print $1' {{ olm_repo_dest }}/deploy/okd/manifests/{{ olm_version }}/30_00-namespace.yaml"
```
Comment (djwhatle) [inline ansible/roles/olm_setup/tasks/main.yml]: Wow, first time I've seen perl run from an ansible playbook... Not sure what to think

**Comment by eriknelson** [comment]
@djwhatle yeah, it's unfortunately the state of things right now. Either we commit some hack workarounds behind switches, or things are just broken and we wait for fixes to trickle down. Also expecting breakages to show up given the rapid pre-alpha development we're playing with so I think it makes sense to switch things to off for now.

**Pull request merged**